ghcide/src/Development/IDE/GHC/Warnings.hs

```diff
@@ -28,9 +29,16 @@ import           Language.LSP.Types                (type (|?) (..))
 withWarnings :: T.Text -> ((ModSummary -> ModSummary) -> IO a) -> IO ([(WarnReason, FileDiagnostic)], a)
 withWarnings diagSource action = do
   warnings <- newVar []
+#if __GLASGOW_HASKELL__ >= 900
+  let newAction :: DynFlags -> WarnReason -> Severity -> SrcSpan -> MsgDoc -> IO ()
+      newAction dynFlags wr _ loc msg = do
+        let prUnqual = alwaysQualify :: PrintUnqualified -- TODO: Do something proper here
+#else
   let newAction :: DynFlags -> WarnReason -> Severity -> SrcSpan -> PprStyle -> SDoc -> IO ()
       newAction dynFlags wr _ loc style msg = do
-        let wr_d = map ((wr,) . third3 (attachReason wr)) $ diagFromErrMsg diagSource dynFlags $ mkWarnMsg dynFlags loc (queryQual style) msg
+        let prUnqual = queryQual style
+#endif
```

Comment:
Log actions no longer gets a `PprStyle` in ghc9, so I'm not sure what the proper thing to do here is. Especially if I am to avoid using `CPP`.